apps/email/js/cards/compose.js

```diff
@@ -361,6 +361,18 @@ return [
       };
     },
 
+    haveAddressesChanged: function() {
+      var addrs = this.extractAddresses();
+      function compareAddresses(a, b) {
+        return a.length == b.length && !a.some(function(e, i) {
+          return typeof b[i].address === 'undefined' || e !== b[i].address;
```

Comment:
I think it is enough here to just return `e !== b[i].address` for this. 

Something I have not checked, just doing code review without running the code, is if we need to test for e.address, for cases where user tapped Compose button, selected addresses via the Contacts + activity, saved the card, then goes to drafts, edits the draft, then goes back -- does this function work in that case?

If it turns out it is best for all addresses coming into this card to be of form {name: ..., address: ...}, then we could look at just creating that object without the name part in the message_reader change where it does the `model.api.parseMailbox(addr).address` normalization.